cpp/models/abm/location.h

```diff
@@ -193,6 +194,20 @@ class Location
         return m_cells;
     }
 
+    /**
+     * get the type of mask that is demanded when entering the location
+     * @return type of the mask 
+     */
+    MaskType get_required_mask() const
```

Comment:
```suggestion
    const MaskType& get_required_mask() const
```

1.) The const only means that the function is const, not the MaskType
2.) objects should not be returned as copies but as references and in this case as a constant reference.